### ISSUE TITLE
Is it deliberate that both .bashrc and .bash_profile are editted?

### ISSUE THREAD

**Issue by 3tilley** [Canop/broot#772]
Lots of .bash_profiles load .bashrc, does the `br` script need to be installed into both? Once one is found, could the other be left? Should it favour .bash_profile from a quick read of [this](https://apple.stackexchange.com/questions/51036/what-is-the-difference-between-bash-profile-and-bashrc) stackoverflow

**Comment by Canop** [comment]
Yes, it's deliberate, it's required by MacOS: https://github.com/Canop/broot/issues/84

What downside is there ?

**Comment by 3tilley** [comment]
Ah ok, I saw the comments and thought it might be to do with Mac. Does that mean if it's just in .bash_profile on mac it would work? Or I guess .profile now that zsh is the default on newer models?

**Comment by Canop** [comment]
> Should br call broot with a certain flag to indicate that it was a shell func and suppress that?

This flag is the `outcmd` launch argument. Broot must be launched with this argument to be able to execute a command from the parent shell. If you run broot from a shell function and it doesn't pass this argument, it's useless.

**Comment by 3tilley** [comment]
I'll leave this question open for the discussion of which rc/profile files should be editted, I've moved the discussion of `br` still asking for setup to #774

**Comment by Canop** [comment]
Answered above: both files are patched because macOS terminals only read .bash_profile, and the source line is idempotent so a profile loading .bashrc is harmless. Closing.
